maven-scm-plugin/src/test/java/org/apache/maven/scm/plugin/ChangeLogMojoTest.java

```diff
@@ -103,7 +103,7 @@ public void testChangeLogWithBadUserDateFormat() throws Exception {
 
             fail("mojo execution must fail.");
         } catch (MojoExecutionException e) {
-            assertTrue(true);
+            assertNotNull(e.getMessage());
```

Comment:
I'm guessing that the true meaning is that `mojo.execute()` throws. And IMO it'd better to assert that that happens. The non null message seems to be secondary.

Reply:
Yes, it's secondary but still not great. The real test is the fail on the previous lines.

Reply:
I'd rather see `fail` there replaced with `assertThrows`. This would be meaningful.

Reply:
It means no more or less than this code does. If you want to send that PR, I'll approve it. Otherwise, let's proceed with this change.

Reply:
Proposed change introduces new constraint for the exception message. It wasn't there, wasn't needed then, and isn't needed now.

I think the always true assertion was added to satisfy checkstyle to avoid empty block.

If this PR gets merged, refactoring to come next, would need to consider that. Not only that the exception is thrown (what is currently checked) but also that the message is not null.

IMO, it's unnecessary and it would be better to migrate to `assertThrows` only.
`assertTrue(true)` could then be eliminated.
`assertNotNull(...)` couldn't.

Reply:
Consider context of the test, not a single line.

This shows, that _refactoring_ is already required, to make test's goal clearer.

> It's meaningless.

It has the meaning of making CS happy.

Reply:
Refactoring is not required. This is a standard and well understood idiom for exactly this case that's been in use for multiple decades.

Reply:
putting in meaningless lines of code to make checkstyle happy is not a good thing.

Reply:
I think using `@Test(expected = MojoExecutionException.class)` would be better choice.

Reply:
There is no `@Test(expected ...)` in JUnit 5 
For me `assertThrows` is the best we control which one action generate exception

Reply:
> There is no `@Test(expected ...)` in JUnit 5

https://github.com/apache/maven-scm/blob/ad63ac59a8dff901da20e9e53adcdcbd4477799f/pom.xml#L281-L285

Reply:
I know that now ju 4 is used ... 😄